string robot_name
float64 concentration_value